src/control/server/faultdomain.go

```diff
@@ -88,3 +83,46 @@ func getFaultDomainFromCallback(callbackPath string) (*system.FaultDomain, error
 
 	return newFaultDomainFromConfig(trimmedOutput)
 }
+
+func checkFaultDomainCallback(path, requiredDir string) error {
+	if path == "" {
+		return errors.New("no callback path supplied")
+	}
+
+	// Must be under the required directory
+	absDir, err := filepath.Abs(requiredDir)
+	if err != nil {
+		panic(err) // shouldn't be possible
```

Comment:
Typically we only want to use `panic()` in cases where it will be caught during development (e.g. `regexp.MustCompile()`, etc.)... Even though this seems like a "can't happen" error, it could possibly be triggered at runtime out in the wild, right? Why not just return the error, perhaps wrapped with a "wow this really shouldn't happen" annotation?

Reply:
I don't think Abs checks that the directory exists, actually, it just formats the string as a plausible path based on the working directory. My assumption was the caller should always pass a string that should be a path, since we control that string. Now that you bring this up, though, I don't think we should make assumptions about the implementation of Abs. Will fix.